### PR TITLE
add `fromSignificandAndExponent` for Fixed6 and Fixed18

### PR DESCRIPTION
**Pull request by prateekdefi**
Issue: https://linear.app/perennial/issue/PE-1154/[root]-make-fixedfromsignificand-exponent-helper

### CODE REVIEW

**Review by EdNoepel — COMMENTED** [review]
Looks good, but why only on the signed types?

**Comment by prateekdefi** [comment]
> Looks good, but why only on the signed types?

I added only for signed types because the issue mentioned `Fixed`. I think @kbrizzle can confirm if the oracle logic requires this function for unsigned as well. I will add for unsigned if required.

**Comment by kbrizzle** [comment]
@prateekdefi Yeah, let's add to `UFixed` as well for completeness even though that won't be directly used in the oracle right away.

**Comment by prateekdefi** [comment]
> @prateekdefi Yeah, let's add to `UFixed` as well for completeness even though that won't be directly used in the oracle right away.

Added for `UFixed` as well

File: contracts/number/types/Fixed6.sol [hunk 1]
```diff
@@ -82,6 +82,16 @@ library Fixed6Lib {
         return roundOut ? Fixed6.wrap(NumberMath.divOut(Fixed18.unwrap(a), 1e12)): from(a);
     }
 
+    /**
+     * @notice Creates a signed fixed-decimal from a significand and an exponent
+     * @param significand Significand
+     * @param exponent Exponent
+     * @return New signed fixed-decimal
+     */
+    function fromSignificandAndExponent(int256 significand, uint256 exponent) internal pure returns (Fixed6) {
```
Comment (kbrizzle) [inline contracts/number/types/Fixed6.sol:91]: let's stick to the pattern of using `from` for all overloads of the from function.

(we still want the custom name in the mocks to for disambiguation).

File: contracts/number/types/Fixed18.sol [hunk 2]
```diff
@@ -72,6 +72,16 @@ library Fixed18Lib {
         return Fixed18.wrap(Fixed6.unwrap(a) * 1e12);
     }
 
+    /**
+     * @notice Creates a signed fixed-decimal from a significand and an exponent
+     * @param significand Significand
+     * @param exponent Exponent
+     * @return New signed fixed-decimal
+     */
+    function from(Fixed18 significand, uint256 exponent) internal pure returns (Fixed18) {
```
Comment (kbrizzle) [inline contracts/number/types/Fixed18.sol:81]: In all cases the exponent case be negative, see reference here:

https://github.com/equilibria-xyz/perennial-v2/blob/main/packages/perennial-oracle/contracts/pyth/PythFactory.sol#L74.

File: contracts/number/types/Fixed18.sol [hunk 3]
```diff
@@ -72,6 +72,19 @@ library Fixed18Lib {
...
+     * @notice Creates a signed fixed-decimal from a significand and an exponent
+     * @param significand Significand
+     * @param exponent Exponent
+     * @return New signed fixed-decimal
+     */
+    function from(Fixed18 significand, int256 exponent) internal pure returns (Fixed18) {
+        if (exponent < 0) {
+            return significand.div(from(int256(10 ** uint256(-1 * exponent))));
+        }
+        return significand.mul(from(int256(10 ** uint256(exponent))));
```
Comment (kbrizzle) [inline contracts/number/types/Fixed18.sol:85]: super nit: for stuff like this where we're returning a value based on a branch, we usually style it using a ternary like this:
```
return exponent < 0
    ? significand.div(from(int256(10 ** uint256(-1 * exponent))))
    : significand.mul(from(int256(10 ** uint256(exponent))));
```

Comment (prateekdefi) [reply contracts/number/types/Fixed18.sol]: Fixed in [1b08a0a](https://github.com/equilibria-xyz/root/pull/108/commits/1b08a0ae47dff8537a434a3147335697aa800d8f)

**Review by kbrizzle — APPROVED** [review]

**Review by EdNoepel — APPROVED** [review]
LGTM

**Pull request merged**